UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -1,587 +1,1316 @@
+
+					LeftFaceHit = false;
+				}
+
+
+				vec.x = (int) Mathf.Round(origin.x) + xSteps;
+				vec.y = (int) Mathf.Round(origin.y) + ySteps;
+
+				vecHit.x = origin.x + (float) RelativeX; //+ offsetX;
+				vecHit.y = origin.y + (float) RelativeY; // + offsetY;
+				//Check point here
+
+				if (LeftFaceHit)
+				{
+					// float TestX = ((vecHit.x - 0.5f) - Mathf.Floor(vecHit.x));
+
+					// if (0.05f < Math.Abs(TestX))
+					// {
+						// Logger.Log("Offsetuntouchx = " + Offsetuntouchx + "\n" + "directionx = " + direction.x + "\n" +
+						           // "Step = " + xSteps + "\n" + "vexinv = " + vexinvX + "\n" + "offset = " + offsetX +
+						           // "\n" + "\n" + "Test =" + TestX + "\n" + "Relative =" + (RelativeX)
+						           // + "\n" + "\n"
+						           // + " direction.x = " +  direction.x + " calculationFloat " + calculationFloat
+						           // + "\n" + " xSteps " + xSteps + " gridOffsetx " + gridOffsetx  +"  vexinvX " + vexinvX);
+
+
+					// }
+				}
+				else
+				{
+					// float Testy = ((vecHit.y - 0.5f) - Mathf.Floor(vecHit.y));
+					// if (0.05f < Math.Abs(Testy))
+					// {
+						// Logger.Log("Offsetuntouchx = " + Offsetuntouchy + "\n" + "directionx = " + direction.y + "\n" +
+						           // "Step = " + ySteps + "\n" + "vexinv = " + vexinvY + "\n" + "offset = " + offsetY +
+						           // "\n" + "\n" + "Test =" + Testy + "\n" + "Relative =" + (RelativeY)
+						           // + "\n" + "\n"
+						           // + " direction.y = " +  direction.y + " calculationFloat " + calculationFloat
+						           // + "\n" + " ySteps " + ySteps + " gridOffsety " + gridOffsety  +"  vexinvY " + vexinvY);
+					// }
+				}
+
+				for (var i = 0; i < LayersValues.Length; i++)
+				{
+					if (LayersValues[i].LayerType == LayerType.Objects) continue;
+					if (LTSUtil.IsLayerIn(layerMask, LayersValues[i].LayerType))
+					{
+						TileLcation = null;
+						lock (PresentTiles)
+						{
+							PresentTiles[LayersValues[i]].TryGetValue(vec, out TileLcation);
+						}
+
+						// var wold = (vecHit.ToWorld(PresentMatrix));
+						// Debug.DrawLine(wold + (Vector3.right * 0.09f), wold + (Vector3.left * 0.09f), Color.green, 30);
+						// Debug.DrawLine(wold + (Vector3.up * 0.09f), wold + (Vector3.down * 0.09f), Color.green, 30);
+
+
+						// if (LeftFaceHit)
+						// {
+							// Debug.DrawLine(wold + (Vector3.up * 4f), wold + (Vector3.down * 4), Color.blue, 30);
+						// }
+						// else
+						// {
+							// Debug.DrawLine(wold + (Vector3.right * 4), wold + (Vector3.left * 4), Color.blue, 30);
+						// }
+
+						// ColorUtility.TryParseHtmlString("#ea9335", out var Orange);
+						// var map = ((Vector3) vec).ToWorld(PresentMatrix);
+						// Debug.DrawLine(map + (Vector3.right * 0.09f), map + (Vector3.left * 0.09f), Orange, 30);
+						// Debug.DrawLine(map + (Vector3.up * 0.09f), map + (Vector3.down * 0.09f), Orange, 30);
+
+						if (TileLcation != null)
+						{
+							Vector2 Normal = Vector2.zero;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'Normal'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6319358)